### PR TITLE
Adding datarobot-bp-workshop

### PR DESCRIPTION
**Pull request by danna-naser**
The first attempt at this recipe is here: https://github.com/conda-forge/staged-recipes/pull/18797/files
The license has been updated to it should be good to merge now

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by danna-naser** [comment]
I agree to maintain

**Comment by danna-naser** [comment]
@conda-forge/help-python this is ready for review

File: recipes/datarobot-bp-workshop/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - pytest
```
Comment (synapticarbors) [inline recipes/datarobot-bp-workshop/meta.yaml:20]: Is `pytest` required to build the package? I see that it's imported within a method, but I don't think it's ever actually run.

File: recipes/datarobot-bp-workshop/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    - python >=3.7
+  run:
```
Comment (synapticarbors) [inline recipes/datarobot-bp-workshop/meta.yaml:22]: It looks like you are listing many more dependencies than are in the `setup.py`, which only specifies:

```
"datarobot>=2.27.0,<3"
"networkx>=2.2"
"graphviz>=0.16,<1"
```

Typically you should only need those plus `python`. I looked in the commits and it looks like you're listing the `datarobot` dependencies, but those should be pulled in `datarobot`:
https://github.com/conda-forge/datarobot-feedstock/blob/main/recipe/meta.yaml#L23-L36

and don't need to be included here.

File: recipes/datarobot-bp-workshop/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - requests-toolbelt >=0.6
+    - trafaret <2.0,>=0.7,!=1.1.0
+    - urllib3 >=1.23
+    - attrs >=19.1.0,<20.0
+    - contextlib2 >=0.5.5
+    - numpy
+    - python-dateutil
+    - pytz
+    - six
+test:
```
Comment (synapticarbors) [inline recipes/datarobot-bp-workshop/meta.yaml:39]: I would add the `pip check` back in.

File: recipes/datarobot-bp-workshop/meta.yaml (hunk 2 above)
Comment (danna-naser) [reply recipes/datarobot-bp-workshop/meta.yaml:21]: I've removed the unneeded dependencies

File: recipes/datarobot-bp-workshop/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  run:
+    - datarobot >=2.27.0,<3
+    - python-graphviz >=0.16,<1
+    - networkx >=2.2
+    - python >=3.7
+
+test:
+  imports:
+    - datarobot_bp_workshop
+  commands:
```
Comment (synapticarbors) [inline recipes/datarobot-bp-workshop/meta.yaml:30]: under `test` you need:

```yaml
commands:
  - pip check
requires:
  - pip
```

File: recipes/datarobot-bp-workshop/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    - pip check
+
+about:
+  home: https://blueprint-workshop.datarobot.com
+  summary: This client library is designed to support building DataRobot blueprints.
+  license: Modified 3-Clause BSD
```
Comment (synapticarbors) [inline recipes/datarobot-bp-workshop/meta.yaml:36]: I think it should maybe be:

```
license: BSD-3-Clause-Modification
```

or `BSD-3-Clause` but please consult https://spdx.org/licenses/ to be sure. You want to list a license from the "identifier" column.

Comment (danna-naser) [reply recipes/datarobot-bp-workshop/meta.yaml:36]: We have the extra line
> THIS SOFTWARE IS PROVIDED WITHOUT ANY LICENSE TO ANY PATENTS OR TRADEMARKS.
that I'm not seeing in the available licenses. Can I call this a proprietary license in this line or some other way to not use one of the existing options?

Comment (synapticarbors) [reply recipes/datarobot-bp-workshop/meta.yaml:36]: Ok, maybe just change it to `Modified BSD-3-Clause`, but @conda-forge/staged-recipes does anyone want to chime in with a recommendation how to handle what is a `BSD-3-Clause` license with one additional line added to it ("THIS SOFTWARE IS PROVIDED WITHOUT ANY LICENSE TO ANY PATENTS OR TRADEMARKS.")?

Comment (BastianZim) [reply recipes/datarobot-bp-workshop/meta.yaml:36]: You can add an exception: https://spdx.org/licenses/exceptions-index.html

But if it's not in there I normally recommend going with _LicenseRef-BSD-3-Clause_ or something similar with _LicenseRef_.

Comment (danna-naser) [reply recipes/datarobot-bp-workshop/meta.yaml]: Thank you both for the help!!

**Pull request merged**